torch/csrc/jit/ir/scope.h

```diff
@@ -50,6 +51,26 @@ struct TORCH_API Scope : public c10::intrusive_ptr_target {
 
 struct Function;
 struct InlinedCallStack;
+struct ModuleInstanceInfo {
+ private:
+  c10::ClassTypePtr module_type_{nullptr};
+  std::string instance_name_;
+
+ public:
+  ModuleInstanceInfo(c10::ClassTypePtr module_type, std::string instance_name);
+  c10::ClassTypePtr class_type() {
+    return module_type_;
+  }
+  std::string instance_name() {
+    return instance_name_;
+  }
+  c10::ClassTypePtr class_type() const {
+    return module_type_;
+  }
+  std::string instance_name() const {
```

Comment:
nit: Possibly consider returning `std::string const&` instead to avoid a copy.

Reply:
Generally agree, but am vary of returning reference to class variable since client code may just do `auto name = ..` and use it or pass as a reference further around. In the meantime original class maybe destroyed or mutated? Or client code itself may mutate class variable if return type is non-const.

Reply:
Looks like a dangling reference could be dangerous. Will use return by value here.